homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -376,6 +376,26 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
             );
         }
         SwarmEvent::IncomingConnection { .. } => {}
+        SwarmEvent::ConnectionEstablished {
+            peer_id, endpoint, ..
+        } => {
+            let behavior = event_handler.swarm.behaviour_mut();
+            // only listener addresses should be added to the routing table.
+            // TODO: add identify here to discover more listen addresses and add those.
+            if endpoint.is_listener() {
+                // ignores if the peer failed or not to be added in the routing table.
+                behavior
+                    .kademlia
+                    .add_address(&peer_id, endpoint.get_remote_address().clone());
```

Comment:
From looking at the libp2p code, we should always add an address to Kademlia upon dialing, as it may come back online later. If the endpoint is a dialer, then we can remove the address.

Reply:
not sure I understand what you mean here. Do you mean we should be adding an address when its a diler as well, in case identify gives us addresses the peer is listening on?

Reply:
No, I mean, upon dialing, we should always add the address and then remove if the endpoint is a dialer. Essentially, we should do this in startup, not within the event.